src/test/java/com/dynatrace/openkit/core/OpenKitImplTest.java

```diff
@@ -0,0 +1,247 @@
+        final OpenKitImpl openKit = new OpenKitImpl(logger, config);
+
+        // asynchronously invokes the OpenKit initialization
+        final ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.submit(new OpenKitRunner(openKit));
+
+        // shut it down after a second
+        Thread.sleep(1000);
+        openKit.shutdown();
+
+        assertThat(openKit.isInitialized(), is(false));
+    }
+
+    private class OpenKitRunner implements Callable<Void> {
+        private final OpenKitImpl openKit;
+
+        public OpenKitRunner(OpenKitImpl openKit) {
+            this.openKit = openKit;
+        }
+
+        @Override
+        public Void call() throws Exception {
+            openKit.initialize();
+            openKit.waitForInitCompletion(); // blocking call
+            return null;
+        }
+    }
+
+    @Test
+    public void testCreateSession() {
+        // create test environment
+        final Logger logger = mock(Logger.class);
+        final Configuration config = createTestConfig();
+        final OpenKitImpl openKit = new OpenKitImpl(logger, config);
+        openKit.initialize();
+
+        // create a valid session
+        final Session session = openKit.createSession("127.0.0.1");
+        assertThat(session, not(nullValue()));
+        assertThat(session, instanceOf(SessionImpl.class));
+        final SessionImpl sessionImpl = (SessionImpl) session;
+        assertThat(sessionImpl.isEmpty(), is(true));
+        assertThat(sessionImpl.getEndTime(), is(-1L));
+    }
+
+    @Test
+    public void testCreateSessionWhenNotInitialized() {
+        // create test environment
+        final Logger logger = mock(Logger.class);
+        final Configuration config = createTestConfig();
+        final OpenKitImpl openKit = new OpenKitImpl(logger, config);
+
+        // User has forgotten to initialize => shall not throw an exception
+        final Session session = openKit.createSession("127.0.0.1");
+        assertThat(session, not(nullValue()));
+        assertThat(session, instanceOf(SessionImpl.class));
+    }
+
+    @Test
+    public void testCreateSessionMultiple() {
+        // create test environment
+        final Logger logger = mock(Logger.class);
+        final Configuration config = createTestConfig();
+        final OpenKitImpl openKit = new OpenKitImpl(logger, config);
+        openKit.initialize();
+
+        // create two valid sessions
+        final Session session1 = openKit.createSession("127.0.0.1");
+        final Session session2 = openKit.createSession("127.0.0.1");
+        assertThat(session1, instanceOf(SessionImpl.class));
+        assertThat(session2, instanceOf(SessionImpl.class));
+        final SessionImpl session1impl = (SessionImpl)session1;
+        final SessionImpl session2impl = (SessionImpl)session2;
+
+        // verify that the two sessions exist and are not the same
+        assertThat(session1impl, not(session2impl));

```

Comment:
I guess it would make sense to test for
``` java
assertThat(session1impl, not(sameInstance(session2impl)));
```
This makes clear, that both variables shall not be the same instance.